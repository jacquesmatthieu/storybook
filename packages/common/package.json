{
  "name": "@arborescence/common",
  "version": "1.0.0",
  "description": "Common code between apps",
  "private": true,
  "main": "./src/common.js",
  "dependencies": {
    "@arborescence/config": "^1.0.0",
    "graphql-tag": "^2.9.2",
    "moment": "^2.22.2"
  }
}
