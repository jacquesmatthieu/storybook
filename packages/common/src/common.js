import urlResolver from './routing/urlResolver';

export * from './graphql/commonQueries';
export * from './graphql/commonMutations';
export * from './graphql/commonFragments';

export {
  urlResolver,
};
