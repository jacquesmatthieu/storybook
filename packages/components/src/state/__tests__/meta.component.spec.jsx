import React from 'react';
import { shallow } from 'enzyme';
import { urlResolver } from '@arborescence/common';

import Meta from '../meta.component';

describe('Meta', () => {
  let props;

  beforeEach(() => {
    props = {
      title: 'foo',
      pathname: '/bar',
    };

    urlResolver.set('baz', { path: '/bar', parents: ['foogere'] });
  });

  const getWrapper = () => shallow(<Meta {...props}>Foo</Meta>);

  it('should render properly', () => {
    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with default app', () => {
    // Given
    props.pathname = '/hello';
    urlResolver.set('qux', { path: '/hello', parents: [] });

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with no title prop', () => {
    // Given
    props.title = undefined;

    // When
    const wrapper = getWrapper();

    // Then
    expect(wrapper).toMatchSnapshot();
  });
});
