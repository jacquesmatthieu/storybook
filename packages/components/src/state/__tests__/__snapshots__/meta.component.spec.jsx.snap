// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meta should render properly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <title>
    foo - 
    Foogère
  </title>
  Foo
</HelmetWrapper>
`;

exports[`Meta should render with default app 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <title>
    foo - 
    Arborescence
  </title>
  Foo
</HelmetWrapper>
`;

exports[`Meta should render with no title prop 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <title>
    Foogère
  </title>
  Foo
</HelmetWrapper>
`;
